app/services/sendgrid_service.rb

```diff
@@ -1,7 +1,13 @@
 class SendgridService
   attr_accessor :user, :template_name, :substitution_hash
 
-  MAIL_METHODS = %w[send_appointment_confirmation_email send_order_submission_email]
+  MAIL_METHODS = [
+    { name: :send_appointment_confirmation_email, template: 'appointment_confirmation' },
+    { name: :send_order_submission_pickup_email, template: 'submission_pickup' },
+    { name: :send_order_submission_delivery_email, template: 'submission_delivery' },
```

Comment:
Same for this one. From order.rb, `send_email_for_order` is called directly. We either should call these methods from the the order model, or should remove these two items from here.

Reply:
@abhinav098  Well I hesitated, it's either one call with passing the argument, or an `if/else` . Have no strong preference. I added them regardless for consistency

Reply:
Let me know if you'd rather have the explicit call

Reply:
I think its better to call the methods explicitly. instead of directly accessing the `send_email_order` method. @patrixr .

Reply:
I would recommend to make this method to `private`.